pombola/south_africa/management/commands/south_africa_import_new_constituency_office_locations.py

```diff
@@ -0,0 +1,138 @@
+
+    def handle(self, *args, **options):
+        if len(args) != 1:
+            raise CommandError(
+                "You must provide the path to the CSV file as an argument."
+                )
+
+        constituency_offices = Organisation.objects.filter(
+            kind__slug='constituency-office')
+
+        desired_storage_path = os.path.join('images', 'constituency-offices')
+        images_directory = os.path.join(
+            settings.MEDIA_ROOT,
+            desired_storage_path
+            )
+        mkdir_p(images_directory)
+
+        with open(args[0]) as csvfile:
+            reader = csv.DictReader(csvfile)
+
+            for row in reader:
+                party, name = (
+                    re.match(
+                        '(.*) Constituency Office (.*)', row['POI_REFERENCE']
+                        ).groups()
+                    )
+                name = re.sub('\s+', ' ', name)
+
+                qs = constituency_offices.filter(
+                    name__regex=r'^{}.+: {}$'.format(party, name))
+
+                try:
+                    org = qs.get()
+                except Organisation.MultipleObjectsReturned:
+                    print "{} Orgs returned: {}".format(
+                        qs.count(),
+                        repr(qs),
+                        )
+                except Organisation.DoesNotExist:
+                    # Fall back to searching for the name and the party in the
+                    # constituency office name
+                    qs = (
+                        Organisation.objects
+                        .filter(kind__slug='constituency-office')
+                        .filter(name__contains=name)
+                        .filter(name__contains=party)
+                        )
+
+                    org = qs.get()
+
+                place = (
+                    org.place_set
+                    .filter(name__contains='Approximate position of')
+                    .filter(kind__slug='constituency-office')
+                    .get()  # There should be only one.
+                    )
+
+                lon, lat = location_re.match(row['Constituency_Pin']).groups()
+
+                place.location = Point(float(lon), float(lat))
+                place.save()
+
+                # Now get the photo
+                photo_url = row.get('Constituency_Photo')
+
+                try:
+                    org.images.get(source=photo_url)
+                    print (
+                        "Skipping {} as url matches existing image."
+                        .format(org.slug)
+                        )
+                    continue
+                except Image.DoesNotExist:
+                    print "Adding new image to {}.".format(org.slug)
+
+                if photo_url:
+                    response = requests.get(photo_url)
+                    image_filename = '{}'.format(org.slug)
+                    image_filepath = os.path.join(
```

Comment:
This will save the image directly to the media root, which potentially might overwrite an existing file. (Presumably nothing else is creating images in that subdirectory, but that's not a safe assumption.) I believe that the right way to do it, which is used in the example I pointed out to you (https://github.com/mysociety/yournextrepresentative/blob/master/candidates/models/popolo_extra.py#L581-L585), is to create a FileSystemStorage object, and call save on it with the "desired filename". It might end up writing the image to a safely found variant of that filename if it already exists.